app-typescript/components/ToastWrapper.tsx

```diff
@@ -0,0 +1,90 @@
+import * as React from 'react';
+import { ToastMessage, MessageProp, IMessageOptions } from './ToastMessage';
+
+type State = {
+  top: Array<IMessageOptions>;
+  bottom: Array<IMessageOptions>;
+  'top-right': Array<IMessageOptions>;
+  'top-left': Array<IMessageOptions>;
+  'bottom-right': Array<IMessageOptions>;
+  'bottom-left': Array<IMessageOptions>;
+};
+
+const firstState: State = {
+  top: [],
+  bottom: [],
+  'top-right': [],
+  'top-left': [],
+  'bottom-right': [],
+  'bottom-left': [],
+};
+
+export default class ToastWrapper extends React.PureComponent<{}, State> {
+  static idCounter = 0;
+
+  state: State = firstState;
+
+  constructor(props: {}) {
+    super(props);
+    this.notify = this.notify.bind(this);
+  }
+
+  notify = (message: MessageProp, options: IMessageOptions) => {
```

Comment:
I'd be good to use `options: Omit<IMessageOptions, 'id'>`. When you use `Omit` it creates another type except it doesn't have a property you specify. It's good in this case for preventing consumers of the component from passing `id` since `id` is an internal implementation-specific thing which you generate in UI framework and it shouldn't be passed from anywhere else.